src/services/refactors/convertToNamedParameters.ts

```diff
@@ -0,0 +1,510 @@
+        }
+        return parameters;
+    }
+
+    function createNewArgument(functionDeclaration: ValidFunctionDeclaration, functionArguments: NodeArray<Expression>): ObjectLiteralExpression {
+        const parameters = getRefactorableParameters(functionDeclaration.parameters);
+        const hasRestParameter = isRestParameter(last(parameters));
+        const nonRestArguments = hasRestParameter ? functionArguments.slice(0, parameters.length - 1) : functionArguments;
+        const properties = map(nonRestArguments, (arg, i) => {
+            const property = createPropertyAssignment(getParameterName(parameters[i]), arg);
+            suppressLeadingAndTrailingTrivia(property.initializer);
+            copyComments(arg, property);
+            return property;
+        });
+
+        if (hasRestParameter && functionArguments.length >= parameters.length) {
+            const restArguments = functionArguments.slice(parameters.length - 1);
+            const restProperty = createPropertyAssignment(getParameterName(last(parameters)), createArrayLiteral(restArguments));
+            properties.push(restProperty);
+        }
+
+        const objectLiteral = createObjectLiteral(properties, /*multiLine*/ false);
+        return objectLiteral;
+    }
+
+    function createNewParameters(functionDeclaration: ValidFunctionDeclaration, program: Program, host: LanguageServiceHost): NodeArray<ParameterDeclaration> {
+        const refactorableParameters = getRefactorableParameters(functionDeclaration.parameters);
+        const bindingElements = map(refactorableParameters, createBindingElementFromParameterDeclaration);
+        const objectParameterName = createObjectBindingPattern(bindingElements);
+        const objectParameterType = createParameterTypeNode(refactorableParameters);
+
+        let objectInitializer: Expression | undefined;
+        // If every parameter in the original function was optional, add an empty object initializer to the new object parameter
+        if (every(refactorableParameters, param => !!param.initializer || !!param.questionToken)) {
+            objectInitializer = createObjectLiteral();
+        }
+
+        const objectParameter = createParameter(
+            /*decorators*/ undefined,
+            /*modifiers*/ undefined,
+            /*dotDotDotToken*/ undefined,
+            objectParameterName,
+            /*questionToken*/ undefined,
+            objectParameterType,
+            objectInitializer);
+
+        if (hasThisParameter(functionDeclaration.parameters)) {
+            const thisParameter = functionDeclaration.parameters[0];
+            const newThisParameter = createParameter(
+                /*decorators*/ undefined,
+                /*modifiers*/ undefined,
+                /*dotDotDotToken*/ undefined,
+                thisParameter.name,
+                /*questionToken*/ undefined,
+                thisParameter.type);
+
+            suppressLeadingAndTrailingTrivia(newThisParameter.name);

```

Comment:
Can you explain why we need to suppress trivia and then copy comments instead of just using `newThisParameter.name` as-is?

Reply:
If we don't suppress trivia, trailing comments will be emitted but leading comments will disappear. In essence, I do not know what behavior to expect from trivia when reusing nodes. So I am suppressing both leading and trailing and explicitly copying the comments.